Task completed: GP-297 Improve attacking state

frnzee opened the pull request:
- changed attacking state logics
- changed character's look direction so that they look from their point of view and not from zero point
- added weapon functionality to once enemy spawner
- added check of sphere collider existing to get rid of errors on play mod stopping

frnzee committed:
Attacking state improvements

- changed attacking state logics
- changed character's look direction so that they look from their point of view and not from zero point
- added weapon functionality to once enemy spawner
- added check of sphere collider existing to get rid of errors on play mod stopping

frnzee committed:
Fixed look direction of enemies and player

- fixed look direction of enemies and player, so that enemies look correctly towards the pleyer, and player look towards the cursor

frnzee committed:
Implemented new HUD

- added new HUD assets and placed it accordingly
- changed existing direct links to a new ones and deleted old HUD game objects

ega-ega commented on Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs:
```diff
@@ -113,6 +119,11 @@ public void ActivateAttackCollider()
 
         public void DeactivateAttackCollider()
         {
+            if (!_sphereCollider)
```
Change _sphereCollider to "this"

ega-ega commented on Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs:
```diff
@@ -105,6 +106,11 @@ private void ProcessRotation(bool isMoving)
 
         public void ActivateAttackCollider()
         {
+            if (!_sphereCollider)
```
Change _sphereCollider to "this"

frnzee replied:
done

ega-ega commented on Assets/Scripts/Gameplay/Enemy/Spawners/OnceEnemySpawner.cs:
```diff
@@ -1,11 +1,32 @@
+        private void SpawnOne()
+        {
+            var enemy = Spawn();
+            enemy.ChangeWeapon(_weaponFactory.Create(_weaponSettings, enemy));
```
Move this behaviour to parent class.

frnzee replied:
done

ega-ega committed:
Merge pull request #97 from LighteatersStudio/feature/GP-289_Implement_HUD_Assets

[WIP] GP-289_Implement_HUD_Assets

frnzee committed:
Refactoring

- moved creation of a weapon to parent class
- Y rotation of enemies is now fixed
- removed all unnecessary code

frnzee committed:
Merge branch 'feature/GP-297_ImproveAttackingState' of https://github.com/LighteatersStudio/TopDownShooter into feature/GP-297_ImproveAttackingState
